modules/standard/GPU.chpl

```diff
@@ -910,21 +959,46 @@ module GPU
          writeln(Arr); // [1, 2, 3, 4, 5]
        }
   */
-  proc gpuSort(ref gpuInputArr : [] uint) {
+  proc gpuSort(ref gpuInputArr : [] ?t) {
     if !here.isGpu() then halt("gpuSort must be run on a gpu locale");
-
     if gpuInputArr.size == 0 then return;
+
+    if CHPL_GPU=="cpu" {
+      use Sort only sort;
+      sort(gpuInputArr);
+      return;
+    }
+
+    extern proc chpl_gpu_can_cub_sort(): bool;
+    if chpl_gpu_can_cub_sort() {
+      gpuCubSort(gpuInputArr);
+      return;
+    }
+
+    fallBackRadixSort(gpuInputArr);
+  }
+
+  private proc fallBackRadixSort(ref gpuInputArr : [] ?t) where isCoercible(t, uint){
     // Based on the inputArr size, get a chunkSize such that numChunks is on the order of thousands
     // TODO better heuristic here?
     var chunkSize = Math.divCeil(gpuInputArr.size, 2000);
     parallelRadixSort(gpuInputArr, bitsAtATime=8, chunkSize, noisy=false, distributed=false);
   }
 
+  private proc fallBackRadixSort(ref gpuInputArr : [] ?t) where !isCoercible(t, uint){
+    compilerError("GPU Based sorting without CUB is only supported for arrays of type uint. Please check your ROCM Version and make sure it's >= 5.0.0");
+  }
   // We no doc it so we can test this independently to simulate all cases that can happen with sort
   @chpldoc.nodoc
-  proc parallelRadixSort(ref gpuInputArr : [] uint, const bitsAtATime : int = 8, const chunkSize : int = 512, const noisy : bool = false,
-                         const distributed : bool = false) { // The last argument is for multi GPU sort that is pending a patch before it can work
-
+  proc parallelRadixSort(ref gpuInputArr : [] ?t, const bitsAtATime : int = 8, const chunkSize : int = 512, const noisy : bool = false,
+                         const distributed : bool = false) where isCoercible(t, uint){ // The last argument is for multi GPU sort that is pending a patch before it can work
```

Comment:
could you reflow this and `fallBackRadixSort` to fit in 80 cols (prefer) or at least 100 (can live with it)?